feat: добавить слои controller, service, jpa, repository, mapper. Реализованы CRUD-операции.

DaryaSerova opened the pull request:
feat: добавить слои controller, service, jpa, repository, mapper. Реализованы CRUD-операции.
feat: добавить exception.
refactor: pom.xml блок build

DaryaSerova committed:
feat: добавить слои controller, service, jpa, repository, mapper. Реализованы CRUD-операции.
feat: добавить exception.
refactor: pom.xml блок build

Solntmore commented on src/main/java/ru/mom/remembers/note/controller/NoteController.java:
```diff
@@ -0,0 +1,49 @@
+
+        log.info("Получение объекта по id = " + id);
+        return noteService.getFullNoteById(id);
+    }
```
Что насчет использования ResponseEntity? Тогда можно самому задавать статус-код ответа.
Создано 201, успешный гет запрос 200, удаление успешное 204 и так далее.
По типу: 
@GetMapping("/{eventId}")
    public ResponseEntity<EventFullDto> getEventById(@PathVariable long eventId, HttpServletRequest request) {
        log.debug("A Get/events/{} request was received. Get event by id", eventId);

        return ResponseEntity.status(HttpStatus.OK).body(eventService.getEventById(eventId, request));
    }

Solntmore commented on src/main/java/ru/mom/remembers/note/controller/NoteController.java:
```diff
@@ -0,0 +1,49 @@
+    private final NoteService noteService;
+
+    @GetMapping("/{id}")
+    public FullResponseNoteDto getFullNote(@PathVariable(name = "id") Long id) {
```
Мне кажется, что имя метода было бы логичнее переделать в getNote, ну и соответственно, чтобы метод сервиса назывался также. Это будет соответствовать неймингу остальных методов, а смысловой нагрузки слово Full не несет, так как метода getShortNote у нас не подразумевается, короткая версия будет использоваться в методе getNotes, который я напишу и он будет выдавать список коротких Note

Solntmore replied:
Можно @PathVariable(name = "id") Long id сократить до @PathVariable Long id, так как у нас совпадает название переменной и названия в URI) и также в остальных методах

Solntmore commented on src/main/java/ru/mom/remembers/note/controller/NoteController.java:
```diff
@@ -0,0 +1,49 @@
+    @GetMapping("/{id}")
+    public FullResponseNoteDto getFullNote(@PathVariable(name = "id") Long id) {
+
+        log.info("Получение объекта по id = " + id);
```
Как-будто было бы логичнее слово объект заменить на конкретное "Заметка". Ну, предположим, у нас будут другие объекты и контроллеры, и для более легкого чтения логов, проще сразу понимать о каком именно типе объекта идет речь

Solntmore commented on src/main/java/ru/mom/remembers/note/service/NoteServiceImpl.java:
```diff
@@ -0,0 +1,92 @@
+    @Override
+    public FullResponseNoteDto getFullNoteById(Long id) {
+
+        var noteOpt = notePersistService.getNoteById(id);
```
Читал где-то, что плохим тоном является не понятные сокращения. Может Opt тут и в других поменять на полное Optional? Или думаешь так норм?

Solntmore commented on src/main/java/ru/mom/remembers/note/service/NoteServiceImpl.java:
```diff
@@ -0,0 +1,92 @@
+    public FullResponseNoteDto updateNote(UpdateRequestNoteDto updateNoteDto, Long id) {
+
+        if (updateNoteDto.getId().equals(id)) {
+            throw new BadRequestException("Bad request body", "Id of note is mistake.");
```
Наверное, логично и в логах, и в исключениях один язык использовать. Ну, либо, все на английском, либо на русском) Лучше, английский

Solntmore reviewed (commented):
Оставил пару заметок, посмотри, если что, обсудим

DaryaSerova committed:
style: изменить язык логирования
style: изменить имя метода getNoteById на getNote NotePersistService
feat: добавить метод existNote NotePersistService - проверка существования заметки в БД
fix: добавить ResponseEntity статус-код ответа
